modules/catalog/src/main/java/org/apache/ignite/internal/catalog/commands/CatalogUtils.java

```diff
@@ -41,6 +50,50 @@ public static TableDescriptor fromParams(int id, CreateTableParams params) {
         );
     }
 
+    /**
+     * Converts CreateIndex command params to descriptor.
+     *
+     * @param id Index id.
+     * @param tableId Table id.
+     * @param params Parameters.
+     * @return Index descriptor.
+     */
+    public static IndexDescriptor fromParams(int id, int tableId, CreateIndexParams params) {
+        params.columns().stream()
+                .filter(Predicate.not(new HashSet<>()::add))
+                .findAny()
+                .ifPresent(colName -> {
+                    throw new IllegalArgumentException("Can't create index on duplicate columns: columnName=" + colName);
```

Comment:
there should be no validation in utility classes. There might be assertion to validate that certain invariants are held, but no for validation of user input.

Besides, we need to stop using for such kind of errors